CMakeLists.txt

```diff
@@ -235,7 +235,7 @@ if(WITH_CUDA)
   endif()
 endif(WITH_CUDA)
 
-find_package(PythonInterp REQUIRED)
+find_package(PythonInterp 3 REQUIRED)
```

Comment:
> At the moment we still allow compilation with Python2 bindings, we just offer no guarantee that it will work. Removing the ability to use Python2 means removing all from __future__ imports, simplifying a few exceptions statements, removing object from classes, simplifying the CMake logic, removing the line you just flagged from GitLab CI and Travis CI files, etc.

How does that work out with that line?

Reply:
This line calls a function that returns early instead of looking for a python interpreter when we pass `-DPYTHON_EXECUTABLE=$(which python2)` to cmake. The build script for CI always passes this variable.

Reply:
that's not true... the cmake script will fail if you call it with ```-DPYTHON_EXECUTABLE=$(which python2)```. That also makes sense since you want to restrict users to python3

Reply:
my bad, I was using the wrong branch to test the CMake behavior. No more Python2 with this line.